### PR TITLE
Bluetooth: Mesh: Energy management sensors test

### PR DESCRIPTION
**Pull request by PavelVPV**
This commits adds unit tests for energy management sensors data
encoding and decoding.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/sensor_types.c [hunk 1]
```diff
@@ -218,11 +246,19 @@ static int scalar_encode(const struct bt_mesh_sensor_format *format,
 	int32_t min_value = scalar_min(format);
 
 	if (raw > max_value || raw < min_value) {
-		uint32_t type_max = BIT64(8 * format->size) - 1;
+		uint32_t type_max;
+
+		if (repr->flags & SIGNED) {
```
Comment (alxelax) [inline subsys/bluetooth/mesh/sensor_types.c:251]: I guess @ludvigsj has already done this :)

File: tests/subsys/bluetooth/mesh/sensor_subsys/src/main.c [hunk 2]
```diff
@@ -148,16 +173,726 @@ static void time_second16_check(const struct bt_mesh_sensor_type *sensor_type)
...
+	}
+}
+
+static void power_specification_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct {
+		struct sensor_value represented[3];
+		struct uint24_t raw[3];
+	} test_vector[] = {
+		{{{0, 0}, {1, 0}, {65536, 0}}, {{0}, {10}, {655360}}},
```
Comment (alxelax) [inline tests/subsys/bluetooth/mesh/sensor_subsys/src/main.c:577]: Could you align this? Looks a bit messy

**Review by alxelax — APPROVED** [review]

File: include/bluetooth/mesh/sensor_types.h [hunk 3]
```diff
@@ -935,27 +953,44 @@ extern const struct bt_mesh_sensor_type bt_mesh_sensor_power_factor;
...
 	bt_mesh_sensor_rel_dev_energy_use_in_a_period_of_day;
 
-/** Relative device runtime in a generic level range
+/** Apparent energy
  *
- *  This sensor type supports series access.
- *  The series X-axis is unitless.
+ *  Channels:
+ *  - Apparent Energy
+ *    - Unit: KVAh
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/sensor_types.h:960]: Correct form appears to be `kVAh`

File: include/bluetooth/mesh/sensor_types.h [hunk 4]
```diff
@@ -935,27 +953,44 @@ extern const struct bt_mesh_sensor_type bt_mesh_sensor_power_factor;
...
  */
-extern const struct bt_mesh_sensor_type
-	bt_mesh_sensor_rel_dev_runtime_in_a_generic_level_range;
+extern const struct bt_mesh_sensor_type bt_mesh_sensor_apparent_power;
+
+/** Active energy loadside
+ *
+ *  Channels:
+ *  - Energy
+ *    - Unit: Kwh
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/sensor_types.h:980]: `kWh`

**Review by trond-snekvik — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]

**Pull request merged**